#ifndef UnpackNetWork_h
#define UnpackNetWork_h

#include <memory>
#include "DEF_ErrorCode.h"
#include "HurtState.h"
#include "DEF_NetMsgId.h"
#include "DEF_CardAttr.h"
#include "DEF_RoleAttr.h"
#include "DEF_LegionAttr.h"
#include "DEF_Configs.h"
#include "DEF_BagTypeDefine.h"
#include "FenjieRequest.pb.h"
#include "FenjieAnswer.pb.h"
#include "MailInsertRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "MailListAnswer.pb.h"
#include "MailOperRequest.pb.h"
#include "MailGetItemAnswer.pb.h"
#include "CommonAnswer.pb.h"
#include "MailOperRequest.pb.h"
#include "MailListAnswer.pb.h"
#include "MailOperRequest.pb.h"
#include "MailSendRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LoginRequest.pb.h"
#include "LoginAnswer.pb.h"
#include "KickRoleNotify.pb.h"
#include "CreateRoleRequest.pb.h"
#include "CreateRoleAnswer.pb.h"
#include "EnterGameRequest.pb.h"
#include "EnterGameAnswer.pb.h"
#include "GetCardAttrRequest.pb.h"
#include "GetCardAttrAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "GetPlayerAttrRequest.pb.h"
#include "GetPlayerAttrAnswer.pb.h"
#include "ListBagRequest.pb.h"
#include "ListBagAnswer.pb.h"
#include "GetPlayerInfoRequest.pb.h"
#include "PlayerInfo.pb.h"
#include "BagNotEnoughNotify.pb.h"
#include "SystemUpdateNotify.pb.h"
#include "RenameRequest.pb.h"
#include "RenameAnswer.pb.h"
#include "HeartbeatRequest.pb.h"
#include "HeartbeatAnswer.pb.h"
#include "GmCmdRequest.pb.h"
#include "GmCmdAnswer.pb.h"
#include "HeroUpgradeLevelRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroUpgradeScalaRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroCheerRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroCheerRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroEmbattleRequest.pb.h"
#include "HeroEmbattleNotify.pb.h"
#include "HeroUpgradeDestinyRequest.pb.h"
#include "HeroUpgradeDestinyAnswer.pb.h"
#include "HeroCultureRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroCultureReplaceRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroUpgradeDestinyRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroInstallAwakenEquipRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroInstallAwakenEquipRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "ItemComposeRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "HeroCheerRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "IncHeroNotify.pb.h"
#include "HeroSellRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "UniteSkillShowNotify.pb.h"
#include "EquipStrengthenRequest.pb.h"
#include "EquipStrengthenAnswer.pb.h"
#include "EquipUpgradeStarRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "EquipRefiningRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "EquipDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "EquipDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "AutoEquipStrengthenRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "EquipSellRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "FightRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RaidRecordAnswer.pb.h"
#include "RaidRewardRequest.pb.h"
#include "RaidRewardAnswer.pb.h"
#include "FriendDekaronRequest.pb.h"
#include "FightAnswer.pb.h"
#include "MainRaidSweepRequest.pb.h"
#include "MainRaidSweepAnswer.pb.h"
#include "RaidResetRequest.pb.h"
#include "RaidResetAnswer.pb.h"
#include "UseItemRequest.pb.h"
#include "UseItemAnswer.pb.h"
#include "SellChipRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "SellAwakenItemRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "FenjieAwakenItemRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "ChipMergeRequest.pb.h"
#include "ChipMergeAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FriendNearAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ArenaGetPlayerAnswer.pb.h"
#include "ArenaStartRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ArenaFlopAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ShopListAnswer.pb.h"
#include "ShopBuyRequest.pb.h"
#include "ShopListAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ArenaGetPlayerAnswer.pb.h"
#include "ArenaMultiStartRequest.pb.h"
#include "ArenaMultiStartAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderInfoAnswer.pb.h"
#include "TowerLadderAttrRequest.pb.h"
#include "TowerLadderInfoAnswer.pb.h"
#include "TowerLadderStartRequest.pb.h"
#include "FightAnswer.pb.h"
#include "TowerLadderEliteRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderInfoAnswer.pb.h"
#include "TowerLadderStartFloorRequest.pb.h"
#include "TowerLadderStartFloorAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TowerLadderSweepAnswer.pb.h"
#include "HuangtingRequest.pb.h"
#include "HuangtingAnswer.pb.h"
#include "ChoukaRequest.pb.h"
#include "ChoukaAnswer.pb.h"
#include "ChoukaActivityNotify.pb.h"
#include "ChoukaShareLuckNotify.pb.h"
#include "DrawLuckPackageRequest.pb.h"
#include "DrawLuckPackageAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelListAnswer.pb.h"
#include "RebelShareRequest.pb.h"
#include "RebelShareAnswer.pb.h"
#include "RebelRaidStartRequest.pb.h"
#include "FightAnswer.pb.h"
#include "RebelHurtRankRequest.pb.h"
#include "RebelRankAnswer.pb.h"
#include "RebelExploitRewardRequest.pb.h"
#include "RebelExploitRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelBossListAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelRankAnswer.pb.h"
#include "RebelRaidStartRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "RebelRankAnswer.pb.h"
#include "RebelShareFlagNotify.pb.h"
#include "RebelCreateFlagNotify.pb.h"
#include "ShenjiangShopRequest.pb.h"
#include "ShenjiangShopBuyRequest.pb.h"
#include "ShenjiangShopAnswer.pb.h"
#include "ShopPlayInfoRequest.pb.h"
#include "ShopPlayInfoAnswer.pb.h"
#include "ShopPlayBuyRequest.pb.h"
#include "ShopPlayBuyAnswer.pb.h"
#include "TreasureDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TreasureDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TreasureStrengthenRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TreasureRefiningRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TreasureForgeRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TreasureSellRequest.pb.h"
#include "CommonSellAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "BlusherListAnswer.pb.h"
#include "BlusherWooRequest.pb.h"
#include "BlusherRewardAnswer.pb.h"
#include "BlusherOperRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "BlusherOperRequest.pb.h"
#include "BlusherRewardAnswer.pb.h"
#include "BlusherOperRequest.pb.h"
#include "BlusherRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "BlusherRankAnswer.pb.h"
#include "ShopBuyRequest.pb.h"
#include "ShopPlayBuyAnswer.pb.h"
#include "ChatSendRequest.pb.h"
#include "ChatSendAnswer.pb.h"
#include "ChatMsgNotify.pb.h"
#include "BroadcastNotify.pb.h"
#include "AnnouncementNotify.pb.h"
#include "ModuleSubscribeRequest.pb.h"
#include "CommonRequest.pb.h"
#include "EatCucumberAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TaskListAnswer.pb.h"
#include "TaskGetRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "TaskGetRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TaskListAnswer.pb.h"
#include "TaskGetRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "TaskGetRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TaskListAnswer.pb.h"
#include "TaskGetRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "SignRequest.pb.h"
#include "SignAnswer.pb.h"
#include "LuxurySignGetRewardRequest.pb.h"
#include "LuxurySignGetRewardAnswer.pb.h"
#include "MoneyTreeRequest.pb.h"
#include "MoneyTreeAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ActivityListAnswer.pb.h"
#include "ActivityRewardRequest.pb.h"
#include "TaskGetRewardAnswer.pb.h"
#include "DrawVipDailyWelfareRequest.pb.h"
#include "DrawVipDailyWelfareAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "OpenSrvRankActivityInfo.pb.h"
#include "OpenSrvRankDrawAwardRequest.pb.h"
#include "OpenSrvRankDrawAwardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "VipWeekPackageInfoAnswer.pb.h"
#include "VipWeekPackageBuyRequest.pb.h"
#include "VipWeekPackageBuyAnswer.pb.h"
#include "BlessingBagNotify.pb.h"
#include "EnterGameGuideNotify.pb.h"
#include "LevelUpNotify.pb.h"
#include "CommonRankRequest.pb.h"
#include "CommonRankAnswer.pb.h"
#include "CommonRankLikeRequest.pb.h"
#include "CommonRankLikeAnswer.pb.h"
#include "ViewPlayerRequest.pb.h"
#include "ViewPlayerAnswer.pb.h"
#include "QiecuoStartRequest.pb.h"
#include "QiecuoStartServerRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FriendListAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendRecvEnergyAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FriendListAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FriendListAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "FriendOperatorRequest.pb.h"
#include "FriendOperatorAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FriendListAnswer.pb.h"
#include "FriendAddByNameRequest.pb.h"
#include "GuildCommitRequest.pb.h"
#include "GuildCommitRequest.pb.h"
#include "UseCDKAnswer.pb.h"
#include "SpecialAwardNotify.pb.h"
#include "LegionCreateRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionMemberListAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionEventAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionListAnswer.pb.h"
#include "LegionApplyRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "LegionApplyRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionApplyListAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "LegionOperateAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "LegionOperateRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "LegionChangeLogoRequest.pb.h"
#include "LegionChangeLogoAnswer.pb.h"
#include "LegionChangeBulletinRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "LegionChangeBulletinRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionMessageAnswer.pb.h"
#include "LegionSendMessageRequest.pb.h"
#include "LegionSendMessageAnswer.pb.h"
#include "LegionStickyMessageRequest.pb.h"
#include "LegionStickyMessageAnswer.pb.h"
#include "LegionStickyMessageRequest.pb.h"
#include "LegionStickyMessageAnswer.pb.h"
#include "LegionWorshipRequest.pb.h"
#include "LegionWorshipAnswer.pb.h"
#include "LegionWorshipOpenBoxRequest.pb.h"
#include "LegionWorshipOpenBoxAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionLimitShopInfo.pb.h"
#include "LegionLimitShopBuyRequest.pb.h"
#include "LegionLimitShopBuyAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionSkillInfoAnswer.pb.h"
#include "LegionSkillOperateRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "LegionSkillOperateRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionRankAnswer.pb.h"
#include "LegionChangeAcceptTypeRequest.pb.h"
#include "LegionChangeAcceptTypeAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "KingFightAnswer.pb.h"
#include "KingFightStageInfoRequest.pb.h"
#include "KingFightStageInfoAnswer.pb.h"
#include "KingFightStartRequest.pb.h"
#include "FightAnswer.pb.h"
#include "KingFightUseItemRequest.pb.h"
#include "KingFightUseItemAnswer.pb.h"
#include "KingFightBuyCntRequest.pb.h"
#include "KingFightBuyCntAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "KingFightRankStarAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "KingFightRankDamageAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CrusadeFortressAllInfoAnswer.pb.h"
#include "CrusadeFortressInfoRequest.pb.h"
#include "CrusadeFortressInfoAnswer.pb.h"
#include "CrusadeFortressFightRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CrusadeFortressOpenBoxRequest.pb.h"
#include "CrusadeFortressOpenBoxAnswer.pb.h"
#include "CrusadeFortressRewardRequest.pb.h"
#include "CrusadeFortressOpenBoxAnswer.pb.h"
#include "CrusadeFortressBattleRecordRequest.pb.h"
#include "CrusadeFortressBattleRecordAnswer.pb.h"
#include "CrusadeFortressReplayRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CrusadeFortressBuyBuffRequest.pb.h"
#include "CrusadeFortressBuyBuffAnswer.pb.h"
#include "KingFightBuyCntRequest.pb.h"
#include "KingFightBuyCntAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "WudiPreselectRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "WudiInfoAnswer.pb.h"
#include "WudiFightRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "WudiHurtRankAnswer.pb.h"
#include "BuyWudiFightCntRequest.pb.h"
#include "BuyWudiFightCntAnswer.pb.h"
#include "BuyWudiFightBuffRequest.pb.h"
#include "BuyWudiFightBuffAnswer.pb.h"
#include "BuyWudiBossBuffRequest.pb.h"
#include "BuyWudiBossBuffAnswer.pb.h"
#include "WudiOpenNotify.pb.h"
#include "CommonRequest.pb.h"
#include "VisitInfoAnswer.pb.h"
#include "VisitStartRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "VisitGetRewardRequest.pb.h"
#include "VisitGetRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "VisitInfoRandomAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "VisitFriendListAnswer.pb.h"
#include "VisitFriendRequest.pb.h"
#include "VisitInfoRandomAnswer.pb.h"
#include "VisitFriendRewardRequest.pb.h"
#include "VisitFriendRewardAnswer.pb.h"
#include "VisitFriendAccelerateRequest.pb.h"
#include "VisitFriendAccelerateAnswer.pb.h"
#include "VisitSaveRouteRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "VisitSaveRouteRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "VisitGetSaveRouteAnswer.pb.h"
#include "VisitOneKeyGetRewardRequest.pb.h"
#include "VisitOneKeyGetRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "OpenFundAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "OpenFundGetRequest.pb.h"
#include "OpenFundGetAnswer.pb.h"
#include "OpenFundGetRequest.pb.h"
#include "OpenFundGetAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "MonthCardInfoAnswer.pb.h"
#include "MonthCardGetRequest.pb.h"
#include "MonthCardGetAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FirstRechargeAwardGetAnswer.pb.h"
#include "RechargeNotify.pb.h"
#include "BuyMonthCardRequest.pb.h"
#include "BuyMonthCardAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossInfoAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "FightAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossGetRewardAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossFirstKillRankAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossTotalStageRankAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossLegionMaxDamageRankAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossMaxDamageRankAnswer.pb.h"
#include "LegionBossCommonRequest.pb.h"
#include "LegionBossKillCountRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "LegionBossTaskInfoAnswer.pb.h"
#include "LegionBossTaskGetRequest.pb.h"
#include "LegionBossGetRewardAnswer.pb.h"
#include "FashionDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "FashionStrengthenRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "TurntableNotify.pb.h"
#include "TurntableStartRequest.pb.h"
#include "TurntableStartAnswer.pb.h"
#include "TurntableRefreshRequest.pb.h"
#include "TurntableRefreshAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TurntableRecordAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "TurntableTaskListAnswer.pb.h"
#include "TurntableTaskRewardRequest.pb.h"
#include "TurntableTaskRewardAnswer.pb.h"
#include "ZhaocaimaoNotify.pb.h"
#include "CommonRequest.pb.h"
#include "ZhaocaimaoStartAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ZhaocaimaoRecordAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "ZhaocaimaoTaskListAnswer.pb.h"
#include "ZhaocaimaoTaskRewardRequest.pb.h"
#include "ZhaocaimaoTaskRewardAnswer.pb.h"
#include "OpenUIReport.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceStateAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceGetInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceGetOpponentsAnswer.pb.h"
#include "CSPointsRaceFightRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CSPointsRaceRevengeRequest.pb.h"
#include "FightAnswer.pb.h"
#include "CSPointsRacePrepareFightRequest.pb.h"
#include "CSPointsRacePrepareFightAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceDefendLogAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CSPointsRaceGetOpponentsAnswer.pb.h"
#include "CSPointsRaceFightBuyRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "FenjieGallantRequest.pb.h"
#include "FenjieGallantAnswer.pb.h"
#include "ComposeGallantRequest.pb.h"
#include "FenjieGallantAnswer.pb.h"
#include "GallantEntryInfoNotify.pb.h"
#include "GallantEntryActivateRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "GallantRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "GallantRaidInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "FightAnswer.pb.h"
#include "LimitedTimeBuyNotify.pb.h"
#include "LimitedTimeBuyRequest.pb.h"
#include "LimitedTimeBuyAnswer.pb.h"
#include "RebateInfoNotify.pb.h"
#include "CommonRequest.pb.h"
#include "RebateReceiveAnswer.pb.h"
#include "PetDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "PetDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "PetDressRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "PetUplevelRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "PetStarUpgradeRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "PetUplevelRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "NianEventNotify.pb.h"
#include "CommonRequest.pb.h"
#include "NianInfoAnswer.pb.h"
#include "NianBuyFireworksRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "NianBuyFireworksRequest.pb.h"
#include "NianUseFireworksAnswer.pb.h"
#include "NianGetRedPacketRequest.pb.h"
#include "NianGetRedPacketAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "NianRankAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "NianTaskListAnswer.pb.h"
#include "NianTaskRewardRequest.pb.h"
#include "NianTaskRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CommonRequest.pb.h"
#include "AchievementTaskListAnswer.pb.h"
#include "AchievementTaskRewardRequest.pb.h"
#include "AchievementTaskRewardAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "AchievementOneKeyRewardAnswer.pb.h"
#include "AchievementNewTaskCompleteNotify.pb.h"
#include "CommonRequest.pb.h"
#include "PetRaidInfoAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "PetRaidDeployAnswer.pb.h"
#include "PetRaidFightRequest.pb.h"
#include "FightAnswer.pb.h"
#include "PetRaidBoxNotify.pb.h"
#include "CommonRequest.pb.h"
#include "CommonAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "PetRaidOneKeyPassAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "PetRaidPassCountRankAnswer.pb.h"
#include "CsRiverLanternFloatRequest.pb.h"
#include "CsRiverLanternFloatAnswer.pb.h"
#include "CommonRequest.pb.h"
#include "CsRiverLanternInfoAnswer.pb.h"
#include "CsRiverLanternNewFloatNotify.pb.h"
#include "CsRiverLanternNewRankNotify.pb.h"
#include "CsRiverLanternSelfRankUpdateNotify.pb.h"
#include "CsRiverLanternOthersGetItemNotify.pb.h"
#include "CommonRequest.pb.h"
#include "CsGrouponInfoAnswer.pb.h"
#include "CsGrouponCountUpdateNotify.pb.h"

//去掉0长度数组的警告
#pragma warning (disable:4200)
#pragma pack(1)
typedef struct st_protoHeader
{
	uint16_t len;
	uint16_t msg_id;
	uint16_t seq;
	uint32_t crc;
	char data[0];
} ProtoHeader;
#pragma pack()

struct ProtoPackTool
{
/** msgId 消息ID，protoBuf 返回的二进制（未解析），len 返回的二进制长度（未解析）,result(解析出来的二进制）   **/
	static VGAME_PROTO_EXPORT void getProtoFromByte(int msgId,const void* protoBuf,int len,char** result);
	
	
	static VGAME_PROTO_EXPORT bool getByteFromProto(int msgId,::google::protobuf::MessageLite* proto,char** byteArray,int &byteArrayLen );
};


#endif // ParseMsg_h__
